### PR TITLE
feat(rpc/block): simplify the block_receipts rpc

### PR DESCRIPTION
**Pull request by jsvisa**
Seems like we don't need to declare `block_and_receipts` as mutable, can make the main code flow more simple to read.

### CODE REVIEW

File: crates/rpc/rpc/src/eth/api/block.rs [hunk 1]
```diff
@@ -63,65 +63,69 @@ where
...
+            self.cache().get_block_and_receipts(block_hash).await?
+        } else {
+            None
+        };
+
+        // If no block and receipts found, return None
+        let (block, receipts) = match block_and_receipts {
+            Some(data) => data,
+            None => return Ok(None),
+        };
```
Comment (mattsse) [inline crates/rpc/rpc/src/eth/api/block.rs:81]: this can be replaced with let Some() else {return Ok(None)}

**Review by mattsse — CHANGES_REQUESTED** [review]
one suggestion

File: crates/rpc/rpc/src/eth/api/block.rs (hunk 1 above)
Comment (jsvisa) [reply crates/rpc/rpc/src/eth/api/block.rs:81]: this looks even more clear, thanks

**Review by mattsse — APPROVED** [review]
nice, this is a lot easier to read now

**Pull request merged**